src/qrscanner.py

```diff
@@ -168,9 +170,33 @@ def update(self):
             if len(res) > 0:
                 self.data += res.decode('utf-8')
             if self.is_done() and self.callback is not None:
+                err_msg = None
                 data = self.data[:-len(self.EOL)]
                 self.reset()
-                self.callback(data)
+                # is this animated QR code
+                if data.startswith("part") and self.callback_animated_qr is not None:
+                    try:
+                        indx = QRAnimated.indx_extract(data)
+                        assert indx[0] <= indx[1]
+
+                        QRAnimated.payload.append(QRAnimated.payload_extract(data))
+                        data = "".join(QRAnimated.payload)
+
+                        if QRAnimated.prev_indx == []:
+                            assert indx[0] == 1
+                        else:
+                            assert indx[0] > QRAnimated.prev_indx[0]
+                            assert indx[1] == QRAnimated.prev_indx[1]
+
+                        self.callback_animated_qr(indx)
+                        QRAnimated.prev_indx = indx
+                        if indx[0] is not indx[1]:
+                            return
+                    except:
+                        # Gui error is thrown in callback
+                        err_msg = "Animated QR: something went wrong!"
+                self.callback(data, err_msg=err_msg)
```

Comment:
err_msg is causing other callbacks to fail, e.g. Add new wallet. Have to come up with sth different. I wonder if i can use sth like catchit decorator...

Reply:
I was thinking about that... We can either apply `@catchit` to `update()` function in `main`, or if it doesn't work for some reason we can separate `usb_host.update()` and `qr_scanner.update()` to a separate function and use `@catchit` there.